src/scripts/main.js

```diff
@@ -1,3 +1,340 @@
+  }
+
+  if (personSalary.value.length < 1 || personSalary.value < 0) {
+    const salaryStyle = form.querySelector('[data-qa="salary"]');
+
+    salaryStyle.style.borderColor = '#e25644';
+
+    setTimeout(() => {
+      salaryStyle.style.borderColor = '';
+    }, 2000);
+
+    pushNotification(450, 20, 'Щось пішло не так',
+      `Дохід не може бути пустим або від'ємним`, 'error');
+
+    return false;
+  }
+
+  pushNotification(450, 20, 'Вау - це успіх',
+    'Нові дані додано до таблиці', 'success');
+
+  return true;
+};
+
+function saveDataToTable(personName, position, office, age, salary) {
+  const tr = document.createElement('tr');
+
+  tr.innerHTML = `
+     <td>${personName.value}</td>
+     <td>${position.value}</td>
+     <td>${office.value}</td>
+     <td>${age.value}</td>
+     <td>$${numberWithComma(salary.value)}</td>
+   `;
+
+  tableBody.append(tr);
+};
+
+function numberWithComma(x) {
+  return x.toString().replace(/\B(?=(\d{3})+(?!\d))/g, ',');
+};
+
+function salaryToNumber(text) {
+  return +text.slice(1).replace(/,/g, '');
+};
+
+function unselectTable() {
+  const trows = tableBody.querySelectorAll('tr');
+
+  trows.forEach((tr) => {
+    tr.classList.remove('active');
+  });
+};
+
+tableBody.addEventListener('click', (e) => {
+  const row = e.target;
+
+  if (!row || !tableBody.contains(row)) {
+    return;
+  }
+
+  unselectTable();
+  row.closest('tr').classList.toggle('active');
+});
+
+tableBody.addEventListener('dblclick', (e) => {
+  const row = e.target;
+
+  if (!row || !tableBody.contains(row)) {
+    return;
+  }
+
+  const trows = tableBody.querySelectorAll('td');
+  const index = Array.from(trows).findIndex(n => n === row) % 5;
+  const tempValue = row.innerText;
+  let input;
+
+  switch (index) {
+    case 2:
+      input = document.createElement('select');
+
+      input.innerHTML = `
+         <select>
+           <option value="Tokyo">Tokyo</option>
+           <option value="Singapore">Singapore</option>
+           <option value="London">London</option>
+           <option value="New York">New York</option>
+           <option value="Edinburgh">Edinburgh</option>
+           <option value="San Francisco">San Francisco</option>
+         </select>
+       `;
+      break;
+    case 3:
+      input = document.createElement('input');
+      input.type = 'number';
+      input.placeholder = 'age value 18 - 90';
+      break;
+    case 4:
+      input = document.createElement('input');
+      input.type = 'number';
+      input.placeholder = 'just a number';
+      break;
+    default:
+      input = document.createElement('input');
+      input.type = 'text';
+  }
+
+  row.innerText = '';
+  row.append(input);
+  input.focus();
+
+  input.onblur = () => {
+    inputValidation(index);
+  };
+
+  input.addEventListener('keydown', (evt) => {
+    if (evt.code !== 'Enter') {
+      return;
+    }
+
+    inputValidation(index);
+  });
+
+  const inputValidation = (i) => {
+    if (input.value !== '') {
```

Comment:
To avoid nesting you can return what you need if you don't have `input.value` in the begining
```
if (!input.value) {
  return row.innerText = tempValue;
}

Reply:
Якщо робити так, то лінтер матюкається